staff/acct/sorry/sorry

```diff
@@ -1,156 +1,202 @@
+SORRY_SHELL = pathlib.Path('/opt/share/utils/bin/sorried')
+# TODO: This sorry log file does not exist anymore, do we still log to file?
+SORRY_LOG = pathlib.Path('/opt/acct/sorry.log')
+kerberos_new_ticket = True
+
+
+def find_sorry_file(file_path):
+    # Own sorry file
+    if os.path.isfile(file_path):
+        return file_path
+
+    # Default sorry files
+    default_sorry_file = os.path.join(REASON_PATH, file_path)
+    if os.path.isfile(default_sorry_file):
+        return default_sorry_file
+
+    raise FileNotFoundError('Sorry file "{}" does not exist.'.format(file_path))
+
+
+def sorry_user(username, sorry_file):
+    attributes = ocflib.account.search.user_attrs(username)
+    if attributes is None:
+        raise ValueError('User {0} not found.'.format(username))
+
+    rootstaffer = real_user()
+    kerberos_new_ticket = kerberos_init(rootstaffer)
+
+    user_dir = attributes['homeDirectory']
+
+    print('Copying sorry file and making .oldshell file')
+
+    if not os.path.isfile(user_dir + '/.oldshell'):
+        with open(user_dir + '/.oldshell', 'w') as oldshell_file:
+            oldshell = attributes['loginShell']
+            print(oldshell, file=oldshell_file)
+
+    shutil.copy2(sorry_file, user_dir + '/.sorry')
+    os.chmod(user_dir + '/.sorry', 0o400)
+    shutil.chown(user_dir + '/.sorry', user=username, group='ocf')
+
+    print("Changing user's shell to sorry shell")
+
+    ocflib.infra.ldap.modify_ldap_entry(ocflib.account.utils.dn_for_username(username),
+                                        {'loginShell': SORRY_SHELL, 'gidNumber': grp.getgrnam('sorry').gr_gid})
+
+    http_dir = ocflib.account.utils.web_dir(username)
+    print('Changing permissions on httpdir to 000')
+    if (os.path.isdir(http_dir)):
+        os.chmod(http_dir, 0o000)
+
+    print('Changing permissions on home directory to 500')
+    os.chmod(user_dir, 0o500)
+
+    # Omitting final system check for now
+    with open(sorry_file, 'r') as filep:
+        log(rootstaffer, username, filep.read())
+        user_mail = ocflib.account.utils.get_email(username)
+        if user_mail:
+            send_sorry_mail(user_mail, username, filep.read())
+        choice = input("Do you want this script to run 'note -u {0} reason' for you? [Y/n]".format(username))
+        if choice not in ['n', 'N']:
+            subprocess.run(['sudo', '-u', rootstaffer, 'note', '-u', format(username), filep.read()])
+        else:
+            print('NOW RUN note -u {0} reason. NOW'.format(username))
+
+    if kerberos_new_ticket:
+        kerberos_destroy()
+
+
+# Returns the real user who is running this script, but probably only works on Linux
+def real_user():
+    if 'SUDO_USER' in os.environ and os.environ['SUDO_USER'] != 'root':
+        return os.environ['SUDO_USER']
+    else:
+        print("The sorry.log is much more useful when it logs who you are rather than simply 'root'.")
+        username = input('Please enter your username: ')
```

Comment:
Perhaps just ask "Please enter your username (for logging purposes): "?

Reply:
That's copied from the original script, but will do